utils/vptr/include/virtual_ptr.hpp

```diff
@@ -247,11 +259,13 @@ class PointerMapper {
    * @param ptr The virtual pointer
    * @param cgh Reference to the command group scope
    */
-  template <sycl_acc_mode access_mode,
-            sycl_acc_target access_target = sycl_acc_target::global_buffer>
+  template <sycl_acc_mode access_mode = default_acc_mode,
+            sycl_acc_target access_target = default_acc_target,
+            typename buffer_data_type = buffer_data_type_t>
```

Comment:
This is a larger interface change request, but if we moved the buffer_data_type to be the first template parameter, we could do stuff like:
```c++
auto p = static_cast<float *>(SYCLMalloc(/* some size*/));
... // in command group scope
auto a = get_access<float>(p, cgh);
```
That way we could default the mode and target most of the time, since it is likely the type that will most often need to change. That said if you don't think it's a good idea, let me know. It's an interface change for the sake of a little typing, so no big deal :)

Reply:
You can provide both versions to avoid breaking existing codes...

Reply:
I see what you mean, Duncan. To be honest I don't know how much existing code that might break is out there.

Ideally, I would like to dispense with needing to pass the data type at all. I have an idea and will try to do this in another MR.